storage/grpc_client.go

```diff
@@ -2084,9 +2131,11 @@ type readResponseDecoder struct {
 	currBuf int    // index of the current buffer being processed
 	currOff uint64 // offset in the current buffer
 	// Processed data
-	msg         *storagepb.BidiReadObjectResponse // processed response message with all fields other than object data populated
-	dataOffsets bufferSliceOffsets                // offsets of the object data in the message.
-	done        bool                              // true if the data has been completely read.
+	msg *storagepb.BidiReadObjectResponse // processed response message with all fields other than object data populated
+	// The original dataOffsets field is replaced with a map to handle multiple ranges.
+	// dataOffsets bufferSliceOffsets
+	dataOffsets map[int64]bufferSliceOffsets // New field: map ReadId to its data offsets
```

Comment:
This is useful for the PR but perhaps confusing for posterity - suggest to remove references to the new vs old field

Reply:
Done.